src/lib.rs

```diff
@@ -1,14 +1,363 @@
-pub fn add(left: u64, right: u64) -> u64 {
-    left + right
+use std::hash::{DefaultHasher, Hash, Hasher};
+
+/// Returns the hash of a single `Hash` value.
+fn hash_single<H: Hash>(value: H) -> u64 {
+    let mut hasher = DefaultHasher::new();
+    value.hash(&mut hasher);
+    hasher.finish()
+}
+
+/// Returns the hash resulting of combining two `Hash` values.
+fn hash_pair<H: Hash>(first: H, second: H) -> u64 {
+    let mut hasher = DefaultHasher::new();
+    first.hash(&mut hasher);
+    second.hash(&mut hasher);
+    hasher.finish()
+}
+
+/// Returns a given index's nth ancestor's index under its respective level.
```

Comment:
The documentation saying `index's index` may be a bit confusing. Also, what is an index in this context?

Reply:
Updated.